### ISSUE TITLE
Why do you divide the gradient by the batch_size_multiplier for ResNet50 training on ImageNet?

### ISSUE THREAD

**Issue by deJQK** [NVIDIA/DeepLearningExamples#801]
Based on my understanding, you are trying to use a smaller batch size (`batch_size * world_size`) to calculate loss and gradient, while update the weight less frequently with `optimizer_batzh_size` to save training effort, and to mimic large batch size training (`optimizer_batzh_size = 2048`) with smaller batch size (`batch_size * world_size = 128 * 8 = 1024`) which is compatible with the device memory. However, I notice you also divide the gradient by the `batch_size_multiplier (= optimizer_batch_size / batch_size / world_size)` [here](https://github.com/NVIDIA/DeepLearningExamples/blob/master/PyTorch/Classification/ConvNets/image_classification/training.py#L257). I wonder if there is any benefit on doing so or some other reason for this. Thanks.

**Comment by asulecki** [comment]
Hi @deJQK, thanks for your question!

Usually, when calculating the loss function, the loss is being averaged over all the elements in a batch.
That means that the gradient you will get after `loss.backward()` will be a mean of gradients of all the examples in this batch.

For the sake of this argument lets assume that we have `optimizer_batch_size=4` with `batch_size=2`
When simulating larger batch sizes, we calculate 2 losses, one for images 1 and 2, second one for images 3 and 4/

What happens is:
 - we zero all the gradients
 - `weight_grad = 0`
 - we calculate the loss for examples 1,2
 - we call `loss.backward()` which adds the gradients of the loss (examples 1,2)  to the (zeroed out) gradient buffer
 - `weight_grad = (grad_img_1 + grad_img_2)/2`
 - we calculate the loss for examples 3,4
 - we call `loss.backward()` which adds the gradients of the loss (examples 3,4) to the gradient buffer
 - `weight_grad = (grad_img_1 + grad_img_2)/2 + (grad_img_3 + grad_img_4)/2 = (grad_img_1 + ... + grad_img_4)/2`
 
 What we really want is `(grad_img_1 + ... + grad_img_4)/4`, so we have to divide the buffer by `batch_size_multiplier` (which is equal to 2 in this case)

**Comment by deJQK** [comment]
Hi @asulecki , thanks a lot for your kind explanation. Now I understand it.
